### ISSUE TITLE
Issues in Firefox

### ISSUE THREAD

**Issue by cdCorey** [ryanburnette/scroll-to-by-speed#6]
The latest version (containing my "context" feature) seems to have some issues with reliably scrolling to the correct position in Firefox. This does not occur in the previous version, so I have to assume my code introduced this bug.

Not sure why this would occur. When no context is specified, the behavior should be _identical_ to the previous behavior. I'll try to take a closer look at this over the weekend.

**Comment by sneppa** [comment]
I fixed the code
Line 32 replace:
    targetPos = Math.abs( $d.scrollTop() + $(this).offset().top - $d.offset().top );
With:
    targetPos = $(this).offset().top;

Works for me in Chrome, Firefox, IE, Safari

**Comment by ryanburnette** [comment]
Could you submit a PR?

**Comment by cdCorey** [comment]
So I wanted to test @sneppa's fix, but when I checked my three original test CodePens ([window](http://codepen.io/cdCorey/full/Iexzc/), [fixed](http://codepen.io/cdCorey/full/vgspz/), [static](http://codepen.io/cdCorey/full/KsLxJ/)) in Chrome, Firefox, and Safari on a Mac, I found that they're all working fine for me. I can't reproduce the issue, so I can't really test a fix for it.

Weird. Am I missing something here? Are those CodePens not working right for you, @ryanburnette?

**Comment by ryanburnette** [comment]
They all work for me now.

**Comment by cdCorey** [comment]
You said you were having issues in Firefox before, and I know one of my
coworkers mentioned something similar. Are you still having issues outside
the CodePen? If so, there may be some other conflict going on that isn't
coming up in the minimum viable test case.

**Comment by ryanburnette** [comment]
The issue might still exist in old versions of Firefox, but I tested it in several up-to-date browsers and all the examples are working now.

**Comment by cdCorey** [comment]
I'm running into problems in Firefox 37.0.2 (the current version). It seems to be a result of $d.scrollTop() returning a different value in Firefox than it does in other browsers (at least Chrome & Safari). In other browsers it always returns 0 when scrolling within the default window context, but in Firefox it returns the same value as $w.scrollTop(). This seems to be the desired result when scrolling within an element context, but results in scrolling too far in window context.

To confirm these results, I tested my minimum viable example CodePens in Firefox 37.0.2:

[Window](http://codepen.io/cdCorey/full/Iexzc/) – NOT WORKING
[Fixed div](http://codepen.io/cdCorey/full/vgspz/) – WORKING
[Static div](http://codepen.io/cdCorey/full/KsLxJ/) – WORKING

**Comment by ryanburnette** [comment]
I looked and the other day I was using Firefox 35.0.1 and all 3 were
working.

**Comment by cdCorey** [comment]
I'm a little concerned about the inconsistent results in different versions of Firefox, but the odds are they all relate to the different value it's returning from that one method. I'm not totally sure why Firefox returns a different value there, but the value from that method is only actually needed within the element context; it's not needed at all for the window context. So the easiest workaround seems to be to simply check which context we're in, and then do different math to set targetPos as appropriate. I've tested this change in Chrome, Firefox, and Safari, and it seems to work great. I've submitted a pull request.
